Automattic/wp-calypso
# Add <ProductCardOptions /> child component and refine CSS

#### Changes proposed in this Pull Request

* Add `ProductCardOptions` sub-component
* Move `PropType` declaration below component definition
* Add tick icon next to the product name if it's purchased
* Add loading state styles, i.e. placeholders for the prices
* Move a SCSS placeholder selectors section to the top of the stylesheet
* Refine `ProductCard`-related styles
* Update devdocs and the readme file

**Mobile**
![Screenshot 2019-10-16 at 21 05 35](https://user-images.githubusercontent.com/478735/66950588-f766ab80-f058-11e9-9ef0-3f239db08601.png)

**Mobile - loading state**
![Screenshot 2019-10-16 at 21 06 02](https://user-images.githubusercontent.com/478735/66950607-00577d00-f059-11e9-9d1f-fda16df4c066.png)

**Desktop**
<img width="702" alt="Screenshot 2019-10-16 at 21 06 31" src="https://user-images.githubusercontent.com/478735/66950632-09484e80-f059-11e9-9f45-75b634c8c7b0.png">

**Desktop - loading state**
![Screenshot 2019-10-16 at 21 06 59](https://user-images.githubusercontent.com/478735/66950643-106f5c80-f059-11e9-8f0c-fd72b08d3cfa.png)

#### Testing instructions

* Go to devdocs and find `ProductCard` UI component
* Confirm that the product options are displayed as in the designs
* Confirm that the product card is responsive and looks as expected on various screen sizes
* Toggle the placeholder state and confirm that the product prices are replaced by a placeholder animation and that this doesn't cause any box dimensions changes (i.e. no unnecessary reflows/repaints)

Fixes n/a
